internal/integration/integration_test.go

```diff
@@ -0,0 +1,392 @@
+		if got, want := len(exposures), 5; got != want {
+			t.Fatalf("expected %v to be %v", got, want)
+		}
+	}
+
+	// Wait for the export to be created and get the list of files
+	var batchFiles []string
+	eventually(t, 30, func() error {
+		// Trigger an export
+		if err := client.ExportBatches(); err != nil {
+			return err
+		}
+
+		// Start export workers
+		if err := client.StartExportWorkers(); err != nil {
+			return err
+		}
+
+		// Attempt to get the index
+		index, err := env.Blobstore().GetObject(ctx, exportDir, "index.txt")
+		if err != nil {
+			if errors.Is(err, storage.ErrNotFound) {
+				return retry.RetryableError(err)
+			}
+			return err
+		}
+
+		// Ensure the new export is created
+		batchFiles = strings.Split(string(index), "\n")
+		switch len(batchFiles) {
+		case 0:
+			return fmt.Errorf("somehow there are no exports?")
+		case 1:
+			return retry.RetryableError(fmt.Errorf("next export does not exist yet"))
+		case 2:
+		default:
+			return fmt.Errorf("expected 2 exports, got %d", len(batchFiles))
+		}
+		return nil
+	})
+
+	// Find the export file that contains the batch - we need this to modify the
+	// batch later to force it to cleanup.
+	exportFile, err := exportdb.New(db).LookupExportFile(ctx, batchFiles[0])
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Mark the export in the past to force a cleanup
+	if err := db.InTx(ctx, pgx.Serializable, func(tx pgx.Tx) error {
+		result, err := tx.Exec(ctx, `
+			UPDATE
+				ExportBatch
+			SET
+				start_timestamp = $1,
+				end_timestamp = $2
+			WHERE
+				batch_id = $3
+		`,
+			time.Now().Add(-30*24*time.Hour),
+			time.Now().Add(-29*24*time.Hour),
+			exportFile.BatchID,
+		)
+		if err != nil {
+			return err
+		}
+		if got, want := result.RowsAffected(), int64(1); got != want {
+			return fmt.Errorf("expected %v to be %v", got, want)
+		}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	// Ensure the export was deleted
+	eventually(t, 30, func() error {
+		// Trigger cleanup
+		if err := client.CleanupExports(); err != nil {
+			return err
+		}
+
+		// Attempt to get the index
+		index, err := env.Blobstore().GetObject(ctx, exportDir, "index.txt")
+		if err != nil {
+			if errors.Is(err, storage.ErrNotFound) {
+				return retry.RetryableError(err)
+			}
+			return err
+		}
+
+		// Ensure the new export is created
+		batchFiles = strings.Split(string(index), "\n")
+		for _, f := range batchFiles {
+			if f != exportFile.Filename {
+				continue
+			}
+
+			// Lookup the file, hope it's gone
+			if _, err := env.Blobstore().GetObject(ctx, exportDir, f); err != nil {
+				if errors.Is(err, storage.ErrNotFound) {
+					return nil // expected
+				} else {
+					return err
+				}
+			}
+
+			return retry.RetryableError(fmt.Errorf("export file still exists"))
+		}
+
+		return nil
+	})
+}
+
+// getExposures finds the exposures that match the given criteria.
+func getExposures(db *database.DB, criteria publishdb.IterateExposuresCriteria) ([]*publishmodel.Exposure, error) {
```

Comment:
I recently added a function in publishdb to read back exposures by key - so if we know what we're looking for we _could_ read it directly